cluster/addons.go

```diff
@@ -226,6 +229,39 @@ func (c *Cluster) doAddonDeploy(ctx context.Context, addonYaml, resourceName str
 	return nil
 }
 
+func (c *Cluster) doAddonDelete(ctx context.Context, resourceName string, isCritical bool) error {
+	k8sClient, err := k8s.NewClient(c.LocalKubeConfigPath, c.K8sWrapTransport)
+	if err != nil {
+		return &addonError{fmt.Sprintf("%v", err), isCritical}
+	}
+	node, err := k8s.GetNode(k8sClient, c.ControlPlaneHosts[0].HostnameOverride)
+	if err != nil {
+		return &addonError{fmt.Sprintf("Failed to get Node [%s]: %v", c.ControlPlaneHosts[0].HostnameOverride, err), isCritical}
+	}
+	deleteJob, err := addons.GetAddonsDeleteJob(resourceName, node.Name, c.Services.KubeAPI.Image)
+	if err != nil {
+		return &addonError{fmt.Sprintf("Failed to generate addon delete job: %v", err), isCritical}
+	}
+	if err := k8s.ApplyK8sSystemJob(deleteJob, c.LocalKubeConfigPath, c.K8sWrapTransport, c.AddonJobTimeout, false); err != nil {
+		return &addonError{fmt.Sprintf("%v", err), isCritical}
+	}
+	// At this point, the addon should be deleted. We need to clean up by deleting the deploy and delete jobs.
+	tmpJobYaml, err := addons.GetAddonsExecuteJob(resourceName, node.Name, c.Services.KubeAPI.Image)
+	if err != nil {
+		return err
+	}
+	if err := k8s.DeleteK8sSystemJob(tmpJobYaml, k8sClient, c.AddonJobTimeout); err != nil {
+		return err
+	}
+
+	if err := k8s.DeleteK8sSystemJob(deleteJob, k8sClient, c.AddonJobTimeout); err != nil {
```

Comment:
curious why do we call the same method twice, with diff param?

Reply:
because we have a k8s job to deploy the addon, and another one to delete it.. when an addon is removed, we delete both jobs to avoid conflict in the future if we want to enable it again..